src/RewardsStreamerMP.sol

```diff
@@ -73,7 +96,91 @@ contract RewardsStreamerMP is UUPSUpgradeable, IStakeManager, TrustedCodehashAcc
         _checkOwner();
     }
 
-    function stake(uint256 amount, uint256 lockPeriod) external onlyTrustedCodehash onlyNotEmergencyMode nonReentrant {
+    /**
+     * @notice Check if a vault is registered
+     * @param vault The address of the vault to check
+     * @return true if the vault is registered, false otherwise
+     */
+    function isVaultRegistered(address vault) public view returns (bool) {
+        return stakeVaultRegistry.vaultOwners(vault) != address(0);
+    }
+
+    /**
+     * @notice Registers a vault with its owner. Called by the vault itself during initialization.
+     * @dev Only callable by contracts with trusted codehash
+     */
+    function registerVault() external onlyTrustedCodehash {
+        address vault = msg.sender;
+        address owner = IStakeVault(vault).owner();
+        stakeVaultRegistry.register(owner, vault);
+    }
+
+    /**
+     * @notice Get all vaults owned by an address
+     */
+    function getUserVaults(address owner) external view returns (address[] memory) {
```

Comment:
I don't think this belongs here, StakeVault should interact directly with Registry, and systems that want to read the registry should interact directly with Registry.

Reply:
Same here, left over! Will remove